File: src/TradingSystem.Web/Services/AiApiService.cs
using System.Net.Http.Json;
using TradingSystem.Web.Models;

namespace TradingSystem.Web.Services;

public interface IAiApiService
{
    Task<List<AiSignalDto>> GetSignalsAsync(string? symbol = null);
    Task<AiSignalDto?> GenerateSignalAsync(string symbol);
    Task<ChatResponse?> SendChatAsync(ChatRequest request);
}

public class AiApiService : IAiApiService
{
    private readonly HttpClient _httpClient;

    public AiApiService(HttpClient httpClient) => _httpClient = httpClient;

    public async Task<List<AiSignalDto>> GetSignalsAsync(string? symbol = null)
    {
        try
        {
            var url = string.IsNullOrEmpty(symbol) ? "/ai/signals" : $"/ai/signals?symbol={symbol}";
            var resp = await _httpClient.GetFromJsonAsync<ApiResponse<List<AiSignalDto>>>(url);
            return resp?.Data ?? new();
        }
        catch { return new(); }
    }

    public async Task<AiSignalDto?> GenerateSignalAsync(string symbol)
    {
        try
        {
            var resp = await _httpClient.PostAsync($"/ai/signals/generate/{symbol}", null);
            var result = await resp.Content.ReadFromJsonAsync<ApiResponse<AiSignalDto>>();
            return result?.Data;
        }
        catch { return null; }
    }

    public async Task<ChatResponse?> SendChatAsync(ChatRequest request)
    {
        try
        {
            var resp = await _httpClient.PostAsJsonAsync("/ai/chat", request);
            var result = await resp.Content.ReadFromJsonAsync<ApiResponse<ChatResponse>>();
            return result?.Data;
        }
        catch { return null; }
    }
}

File: src/TradingSystem.Angular/src/app/services/ai-api.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, of } from 'rxjs';
import { map, catchError } from 'rxjs/operators';
import { ApiResponse, AiSignalDto, ChatRequest, ChatResponse } from '../models/models';

@Injectable({ providedIn: 'root' })
export class AiApiService {
  constructor(private http: HttpClient) {}

  getSignals(symbol?: string): Observable<AiSignalDto[]> {
    const url = symbol ? `/ai/signals?symbol=${symbol}` : '/ai/signals';
    return this.http.get<ApiResponse<AiSignalDto[]>>(url).pipe(
      map(r => r.data || []),
      catchError(() => of([]))
    );
  }

  generateSignal(symbol: string): Observable<AiSignalDto | null> {
    return this.http.post<ApiResponse<AiSignalDto>>(`/ai/signals/generate/${symbol}`, null).pipe(
      map(r => r.data || null),
      catchError(() => of(null))
    );
  }

  sendChat(request: ChatRequest): Observable<ChatResponse | null> {
    return this.http.post<ApiResponse<ChatResponse>>('/ai/chat', request).pipe(
      map(r => r.data || null),
      catchError(() => of(null))
    );
  }
}
